@xen-orchestra/web-core/lib/components/UiLegend.vue

```diff
@@ -0,0 +1,97 @@
+<!-- v1.0 -->
+<template>
+  <li class="legend">
+    <span class="circle" :class="circleColorClass"></span><span class="label typo p3-regular">{{ label }}</span
+    ><UiIcon
+      v-tooltip="{
+        placement: 'top',
+        content: tooltip ? $t(tooltip) : '',
+      }"
+      :icon="tooltip ? faCircleInfo : undefined"
+      color="info"
+      class="ui-icon"
+    />
```

Comment:
- It's preferable to add a `v-if="tooltip"` instead of switching the icon value, which will make the intention clearer.
- A component should not have the responsibility to translate a string prop. It's up to the parent.
- Tooltips are displayed on top by default, so you can pass the value directly to the directive.
- Use meaningful class names.

```suggestion
    <UiIcon v-if="tooltip" v-tooltip="tooltip" :icon="faCircleInfo" class="tooltip-icon" color="info" />
```